ui/src/alerting/components/builder/CheckMatchingRulesCard.tsx

```diff
@@ -1,19 +1,86 @@
 // Libraries
-import React, {FunctionComponent} from 'react'
+import React, {FunctionComponent, useState, useEffect} from 'react'
+import {connect} from 'react-redux'
+import {get} from 'lodash'
 
 // Components
+import MatchingRuleCard from 'src/alerting/components/builder/MatchingRuleCard'
+
+// Actions
+import {getActiveTimeMachine} from 'src/timeMachine/selectors'
+
+// API
+import * as api from 'src/client'
+
+//Types
+import {NotificationRule, Check, AppState, CheckTagSet} from 'src/types'
 import {EmptyState, ComponentSize} from '@influxdata/clockface'
 
-const CheckMatchingRulesCard: FunctionComponent = () => {
+interface StateProps {
+  check: Partial<Check>
+  orgID: string
+}
+
+const CheckMatchingRulesCard: FunctionComponent<StateProps> = ({
+  check,
+  orgID,
+}) => {
+  const getMatchingRules = async (): Promise<NotificationRule[]> => {
```

Comment:
wanna extract this somewhere else, and pass in the `check` and `orgID` as arguments? I think it would be nice to keep it separate from the component / view logic, since it is complex enough.

Reply:
I need to call a function that can call `setMatchingRules` because I can't use an async function within useEffect, and it feels a little awkward to pass the `setMatchingRules` to a function extracted to somewhere else.

Reply:
oh I see, ok makes sense